pkgs/applications/misc/j4-dmenu-desktop/default.nix

```diff
@@ -1,14 +1,14 @@
-{ lib, stdenv, fetchFromGitHub, cmake, dmenu }:
+{ lib, stdenv, fetchFromGitHub, cmake, dmenu, fmt, spdlog }:
 
-stdenv.mkDerivation (finalAttrs: {
+stdenv.mkDerivation rec {
```

Comment:
Why change from finalAttrs to rec?

Reply:
It just felt cleaner. AFAICT the only reason `finalAttrs` was brought in historically was since a non-release revision was selected, but I've switched back to tracking tagged versions.

Reply:
`finalAttrs` generally has better properties, like consistent version overrides (see https://github.com/NixOS/nixpkgs/issues/293452#issuecomment-1979152319 for an example).

Reply:
@jopejoe1 @wegank Thanks for the enlightenment, I've reverted back to using `finalAttrs`.